rasa_nlu/extractors/crf_entity_extractor.py

```diff
@@ -35,13 +35,20 @@ class CRFEntityExtractor(EntityExtractor):
 
     provides = ["entities"]
 
-    requires = ["spacy_doc"]
-
-    function_dict = {'low': lambda doc: doc[0].lower(), 'title': lambda doc: str(doc[0].istitle()),
-                     'word3': lambda doc: doc[0][-3:], 'word2': lambda doc: doc[0][-2:],
-                     'pos': lambda doc: doc[1], 'pos2': lambda doc: doc[1][:2],
-                     'bias': lambda doc: 'bias', 'upper': lambda doc: str(doc[0].isupper()),
-                     'digit': lambda doc: str(doc[0].isdigit())}
+    requires = ["spacy_doc", "tokens"]
+
+    function_dict = {
+        'low': lambda doc: doc[0].lower(),
+        'title': lambda doc: str(doc[0].istitle()),
+        'word3': lambda doc: doc[0][-3:],
+        'word2': lambda doc: doc[0][-2:],
+        'pos': lambda doc: doc[1],
+        'pos2': lambda doc: doc[1][:2],
+        'bias': lambda doc: 'bias',
+        'upper': lambda doc: str(doc[0].isupper()),
+        'digit': lambda doc: str(doc[0].isdigit()),
+        'pattern': lambda doc: doc[2],
```

Comment:
so just a binary feature to indicate that it matched one of the patterns, right?